lib/mail/message.ex

```diff
@@ -316,4 +325,18 @@ defmodule Mail.Message do
 
     mimetype_fn.(extension)
   end
+
+  defp do_get_attachments(messages, attachments) do
+    List.wrap(messages)
+    |> Enum.reduce(attachments, fn
+      %Mail.Message{multipart: true} = message, acc ->
+        do_get_attachments(message.parts, acc)
+      %Mail.Message{multipart: false} = message, acc ->
```

Comment:
I think it would be better to handle this a the function signature level:

```elixir
def get_attachment(%Mail.Message{multipart: false} = message) do
   # single part handling
end

def get_attachments(%Mail.Message{multipart: true} = message) do
  # multipart handling
end
```